File: app/src/main/java/com/filantrop/pvnclient/database/model/FptnServerDto.java
package com.filantrop.pvnclient.database.model;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.io.Serializable;

import lombok.Data;

@Entity(tableName = "server_table")
@Data
public class FptnServerDto implements Serializable {

    public static final FptnServerDto AUTO = new FptnServerDto("Auto", "Auto", "Auto", "", 0);

    @PrimaryKey(autoGenerate = true)
    private int id;
    private String name;

    private String username;
    private String password;
    private String host;
    private Integer port;

    public FptnServerDto(String name, String username, String password, String host, Integer port) {
        this.name = name;
        this.username = username;
        this.password = password;
        this.host = host;
        this.port = port;
    }

    public String getServerInfo() {
        return name + " (" + host + ")";
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getUsername() {
        return username;
    }

    public String getPassword() {
        return password;
    }

    public String getHost() {
        return host;
    }

    public Integer getPort() {
        return port;
    }

    public void setId(int id) {
        this.id = id;
    }
}

File: app/src/main/java/org/fptn/vpnclient/database/model/FptnServerDto.kt
package org.fptn.vpnclient.database.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import org.fptn.vpnclient.core.common.Constants.SELECTED_SERVER_ID_AUTO

@Entity(tableName = "server_table")
data class FptnServerDto(
    @JvmField
    @PrimaryKey(autoGenerate = true)
    val id: Int,
    @JvmField
    val isSelected: Boolean,
    @JvmField val name: String,
    @JvmField val username: String,
    @JvmField val password: String,
    @JvmField val host: String,
    @JvmField val port: Int,
) {
    val serverInfo: String
        get() = "$name ($host)"

    companion object {
        @JvmField
        val AUTO: FptnServerDto =
            FptnServerDto(SELECTED_SERVER_ID_AUTO, false, "Auto", "Auto", "Auto", "", 0)
    }
}
